gpflow/likelihoods/base.py

```diff
@@ -286,7 +286,71 @@ def _variational_expectations(self, Fmu, Fvar, Y):
         raise NotImplementedError
 
 
-class ScalarLikelihood(Likelihood):
+class QuadratureLikelihood(Likelihood):
```

Comment:
Is the code in here specific to how the ScalarLikelihood (now-sub)class works? would it work with another type of likelihood as well (e.g. Heteroskedastic)?

Reply:
The `QuadratureLikelihood` class is an abstract class intended to componentize (any) quadrature object through the implementation of the property `quadrature`, as is done in `NDiagGHQuadratureLikelihood`. It serves the purpose of implementing the 3 main expectation-related methods for likelihoods using the componentized quadrature object, i.e.:
- `_predict_mean_and_var`
- `_predict_log_density`
- `_variational_expectations`
This is, in a way, what `ScalarLikelihood` implemented through `ndiagquad`.

My intention with the QuadratureLikelihood, however, is to have it working with "any" other likelihoods, but I did not try it yet.

Reply:
I think we can simplify the design a little bit (and introduce less classes) by simply having quadrature default to the implementation in `NDiagGHQuadratureLikelihood`. If people want a special/another type of quadrature, they can always subclass and overwrite the `quadrature` property.

Reply:
Done!